dftimewolf/lib/processors/turbinia_gcp_tam.py

```diff
@@ -0,0 +1,361 @@
+      project (str): name of the GCP project containing the disk to process.
+      turbinia_zone (str): GCP zone in which the Turbinia server is running.
+      sketch_id (int): The Timesketch sketch ID.
+      run_all_jobs (bool): Whether to run all jobs instead of a faster subset.
+    """
+
+    self.turbinia_config_file = turbinia_config_file
+
+    for disk in disks.split(','):
+      container = containers.GCEDisk(disk)
+      self.state.StoreContainer(container)
+
+    try:
+      self.TurbiniaSetUp(project, turbinia_zone, sketch_id, run_all_jobs)
+    except TurbiniaException as exception:
+      self.ModuleError(str(exception), critical=True)
+      return
+
+  # pylint: disable=arguments-renamed
+  def Process(self, disk_container: containers.GCEDisk) -> None:
+    """Process a disk with Turbinia."""
+    log_file_path = os.path.join(self._output_path, 'turbinia-{0:s}.log'\
+        .format(disk_container.name))
+    print('Turbinia log file: {0:s}'.format(log_file_path))
+
+    evidence_ = evidence.GoogleCloudDisk(
+        disk_name=disk_container.name,
+        project=self.project,
+        zone=self.turbinia_zone)
+
+    task_data = self.TurbiniaProcess(evidence_)
+
+    local_paths, gs_paths = self._DeterminePaths(task_data)
+
+    if not local_paths and not gs_paths:
+      self.ModuleError(
+          'No interesting files found in Turbinia output.', critical=True)
+
+    timeline_label = '{0:s}-{1:s}'.format(self.project, disk_container.name)
+    # Any local files that exist we can add immediately to the output
+    all_local_paths = [
+        (timeline_label, p) for p in local_paths if os.path.exists(p)]
+
+    downloaded_gs_paths = self._DownloadFilesFromGCS(timeline_label, gs_paths)
+    all_local_paths.extend(downloaded_gs_paths)
+    self.logger.info('Collected {0:d} results'.format(len(all_local_paths)))
+
+    if not all_local_paths:
+      self.ModuleError('No interesting files could be found.', critical=True)
+
+    container: Union[containers.File, containers.ThreatIntelligence]
+    for description, path in all_local_paths:
+      if path.endswith('BinaryExtractorTask.tar.gz'):
+        self.logger.success(
+            'Found BinaryExtractorTask result: {0:s}'.format(path))
+        container = containers.ThreatIntelligence(
+            name='BinaryExtractorResults', indicator=None, path=path)
+      if path.endswith('hashes.json'):
+        self.logger.success('Found hashes.json: {0:s}'.format(path))
+        container = containers.ThreatIntelligence(
+            name='ImageExportHashes', indicator=None, path=path)
+      if path.endswith('.plaso'):
+        self.logger.success('Found plaso result: {0:s}'.format(path))
+        container = containers.File(name=description, path=path)
+      self.state.StoreContainer(container)
+
+  @staticmethod
+  def GetThreadOnContainerType() -> Type[interface.AttributeContainer]:
+    return containers.GCEDisk
+
+  def GetThreadPoolSize(self) -> int:
```

Comment:
Well, seeing it here again I'm guessing this is an override of the parent class. I'm wondering if it would be a good idea to move that `5` value to a constant defined somewhere (perhaps a shared config file if that already exists) so that we can easily adapt this in the future.

Reply:
As mentioned for the other module - The value here is dependant on what the threaded module is actually doing, and is up to the implementation to decide what size thread pool it should support.